source/level/TestGenState.hx

```diff
@@ -22,55 +18,64 @@ class TestGenState extends FlxState {
 	private var map:FlxTilemap;
 
 	private var playerTank:Tank;
+	private var dynamicTank:Tank;
+	private var pursuitTank:Tank;
 	private var enemyTanks:FlxTypedGroup<Tank>;
 
-	private var map:FlxTilemap;
-
-	private var targetTank:Tank;
-	private var enemyTank:Tank;
-
 	override public function create() {
 		super.create();
 
 		var bitmap = LevelGenerator.generateLevel(LEVEL_HEIGHT, LEVEL_WIDTH);
 
 		map = new FlxTilemap();
-		trace(bitmap);
 		map.loadMapFrom2DArray(bitmap, AssetPaths.Wall__png, TILE_WIDTH, TILE_HEIGHT);
 
 		add(map);
-
-		setUpHorizontalTank();
-		targetTank = setUpPlayerTank();
-		enemyTank = setUpDynamicTank(targetTank, map);
+		addTanks();
 	}
 
-	private function setUpHorizontalTank() {
-		var tank = new Tank(500, 50);
-		tank.setControllers(new HorizontalMoveController(tank), new SpinShootController(tank));
-		add(tank.getAllSprites());
-		return tank;
-	}
+	override public function update(elapsed:Float) {
+		super.update(elapsed);
 
-	private function setUpPlayerTank() {
-		var playerTank = TankFactory.NewPlayerTank(100, 50);
-		add(playerTank.getAllSprites());
-		return playerTank;
+		// if (FlxG.keys.justPressed.K) {
+		// 	enemyTanks.getFirstAlive().kill();
+		// }
+
+		FlxG.collide(playerTank, map);
+		FlxG.collide(enemyTanks, map);
+		FlxG.collide(dynamicTank, map);
+		FlxG.collide(pursuitTank, map);
+
+		checkFinished();
 	}
 
-	private function setUpDynamicTank(target, tileMap) {
-		var dynamicTank = new Tank(50, 500);
-		dynamicTank.setControllers(new DynamicMovementController(dynamicTank, target, tileMap),
-			new AutoAimController(dynamicTank, target));
+	private function addTanks() {
+		var tankCoordinates = [250, 300, 350];
+		playerTank = TankFactory.NewPlayerTank(500, 500);
+		dynamicTank = TankFactory.NewDynamicTank(50, 100, playerTank, map);
+		pursuitTank = TankFactory.NewPursuitTank(200, 400, playerTank, map);
+
+		enemyTanks = new FlxTypedGroup<Tank>(3);
+		for (x in tankCoordinates) {
+			var enemy = TankFactory.NewDumbTank(x, 50);
+			enemyTanks.add(enemy);
+		}
+
+		add(playerTank.getAllSprites());
 		add(dynamicTank.getAllSprites());
-		return dynamicTank;
+		trace("HI");
+		add(pursuitTank.getAllSprites());
```

Comment:
To simplify this stuff, I'd recommend removing the `3` from 
```Haxe
enemyTanks = new FlxTypedGroup<Tank>(3);
```
and just add all of the different enemy-type tanks to the `enemyTanks` group. (the three restricts the maximum size of the FlxTypedGroup, I believe. Without it, we can just keep adding stuff to it no problem)

Then you don't need to do a bunch of piecemeal `FlxG.collide()` calls like we do on lines 46 and 47 - everything can be in `enemyTanks` and all of the `enemyTanks` are collided aginst the map.

Not a requirement to fix, since this isn't intended to be the "final" state we use for gameplay, but I think it's good to get comfortable with the more-correct ways for us to handle grouping and adding tanks to our state.

Also, we've got a rogue 
```Haxe
trace("HI");
```
that should probably be deleted.